test/integration/typescript/pages/ssg/blog/[post].tsx

```diff
@@ -1,46 +1,45 @@
 import {
-  InferGetStaticPropsType,
-  GetStaticPaths,
   GetStaticPropsContext,
+  InferGetStaticPathsType,
+  InferGetStaticPropsType,
 } from 'next'
 
 type Post = {
-  author: string
+  id: string
+  title: string
   content: string
 }
 
-export const getStaticPaths: GetStaticPaths = async () => {
+export const getStaticPaths = async () => {
   return {
     paths: [{ params: { post: '1' } }],
     fallback: false,
   }
 }
 
 export const getStaticProps = async (
-  ctx: GetStaticPropsContext<{ post: string }>
+  ctx: GetStaticPropsContext<InferGetStaticPathsType<typeof getStaticPaths>>
```

Comment:
I'm really not sure of how is this better 🤔. Inferring types looks more confusing than actually using a type.